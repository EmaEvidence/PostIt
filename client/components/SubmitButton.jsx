import React from 'react';
import PropTypes from 'prop-types';

/**
 * SubmitButton
 * @method SubmitButton
 *
 * @param {function} props creates the page footer
 *
 * @return {ReactElement} markup
 */
const SubmitButton = props => (
  <input
    type={props.type ? props.type : 'submit'}
    className="form-control btn deep-purple lighten-3 custombutton"
    value={props.value}
  />
  );
SubmitButton.defaultProps = {
  type: 'submit'
};
SubmitButton.propTypes = {
  value: PropTypes.string.isRequired,
  type: PropTypes.string
};
export default SubmitButton;
