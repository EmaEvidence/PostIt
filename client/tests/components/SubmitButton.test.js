import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SubmitButton from '../../components/SubmitButton';

describe('Notification component should', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<SubmitButton />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
