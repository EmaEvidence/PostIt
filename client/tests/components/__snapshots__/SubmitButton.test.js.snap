// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification component should renders without crashing 1`] = `
<input
  className="form-control btn deep-purple lighten-3 custombutton"
  type="submit"
/>
`;
